### ISSUE TITLE
Add `startDate`, `endDate` and `type` params to the list of transactions

### ISSUE THREAD

**Issue by dezoxel** [lunches-platform/api#107]
- User should be able to review the whole list of transaction for all the time. For this reason UI needs start/end date params.
- `type` param is required when user wants to filter his transactions by type

**Comment by warlock39** [comment]
@dezoxel 
**Example of API request**

```
GET /transactions?clientId=349&type=income&startDate=2016-08-01&endDate=2016-08-26 HTTP/1.1
Content-Type: application/json
Host: api.qa.lunches.com.ua
```

**Comment by warlock39** [comment]
@Kostya47 , it can be tested

**Comment by Kostya47** [comment]
@warlock39 I confirm operability, but there is one comment:
We have such request:
`GET /transactions?clientId=348&type=income&startDate=2016-08-31&endDate=2016-09-08`

with response:

```
[{
    "id": "b4aa67d6-ef39-4d96-a36d-beccdb79df1b",
    "type": "income",
    "amount": 200,
    "created": "2016-08-31 15:21:57"
}, {
    "id": "5829c4bb-afe4-444e-9a43-2fea64a7e2c0",
    "type": "income",
    "amount": 200,
    "created": "2016-08-31 15:21:45"
}]
```

And another request with the changed date for the day ahead:
`GET /transactions?clientId=348&type=income&startDate=2016-08-31&endDate=2016-09-09`

with response:

```
[{
    "id": "f042c49d-403a-45ed-93a9-02fe3670e6b9",
    "type": "income",
    "amount": 2500,
    "created": "2016-09-08 09:33:19"
}, {
    "id": "b4aa67d6-ef39-4d96-a36d-beccdb79df1b",
    "type": "income",
    "amount": 200,
    "created": "2016-08-31 15:21:57"
}, {
    "id": "5829c4bb-afe4-444e-9a43-2fea64a7e2c0",
    "type": "income",
    "amount": 200,
    "created": "2016-08-31 15:21:45"
}]
```

As we can see, the first request does not include the transaction on 2016-09-08, I think this is not entirely logical. If you think it is correct operation, then it is possible to close this issue.

**Comment by warlock39** [comment]
@Kostya47 , good point. Lets think about valid pagination by date. The main problem in implementation is data duplication on boundaries (start and end date), we definitely need to not include one of boundary, and common approach is not include right boundary. 
Considering this, valid behaviour looks like this: `[2016-08-31, 2016-09-08)` - include left part and exclude right part.
